x/genutil/utils.go

```diff
@@ -45,13 +47,13 @@ func ExportGenesisFileWithTime(genFile, chainID string, validators []cmttypes.Ge
 }
 
 // InitializeNodeValidatorFiles creates private validator and p2p configuration files. Key type is ed25519.
-func InitializeNodeValidatorFiles(config *cfg.Config) (nodeID string, valPubKey cryptotypes.PubKey, err error) {
-	return InitializeNodeValidatorFilesFromMnemonic(config, "")
+func InitializeNodeValidatorFiles(config *cfg.Config, keyType string) (nodeID string, valPubKey cryptotypes.PubKey, err error) {
+	return InitializeNodeValidatorFilesFromMnemonic(config, "", keyType)
 }
 
 // InitializeNodeValidatorFilesFromMnemonic creates private validator and p2p configuration files using the given mnemonic.
 // If no valid mnemonic is given, a random one will be used instead. Key type is ed25519.
-func InitializeNodeValidatorFilesFromMnemonic(config *cfg.Config, mnemonic string) (nodeID string, valPubKey cryptotypes.PubKey, err error) {
+func InitializeNodeValidatorFilesFromMnemonic(config *cfg.Config, mnemonic string, keyType string) (nodeID string, valPubKey cryptotypes.PubKey, err error) {
```

Comment:
We should keep the `InitializeNodeValidatorFilesFromMnemonic` and `InitializeNodeValidatorFiles` functions the same.

And implement new functions `InitializeNodeValidatorFilesWithKeyType()` that take the extra keytype argument.  I don't want us breaking public API surface with this change